### ISSUE TITLE
remote control interface with dedupe

### ISSUE THREAD

**Issue by mtnjustme** [rclone/rclone#4657]
The dedupe command does support the rc flag, however, it does a lot of stuff before starting to report that it checked/deleted files, and until then the rc stats are 0 checks/deletes, would be nice to have a more live-like rc regarding the dedupe command, also the -P is useless most of the time with dedupe (until it starts to report checks/deletes, which it doesn't at first, but only as it's about to finish).

**Comment by mtnjustme** [comment]
Also it's possible to make dedupe a flag in functions like copy/sync/move, either that or making it possible to copy/sync/move folders that share the same name.

**Comment by ncw** [comment]
> The dedupe command does support the rc flag, however, it does a lot of stuff before starting to report that it checked/deleted files, and until then the rc stats are 0 checks/deletes, would be nice to have a more live-like rc regarding the dedupe command, also the -P is useless most of the time with dedupe (until it starts to report checks/deletes, which it doesn't at first, but only as it's about to finish).

I guess I've always thought of dedupe as more of an interactive tool...

If it counted up directory listing entries would that be useful? As that is what it is doing at the start.

> Also it's possible to make dedupe a flag in functions like copy/sync/move, either that or making it possible to copy/sync/move folders that share the same name.

I'm pretty sure there is an issue about that already...

> also possible to make a new variable "Merges" for example...

Do you mean a statistic to show in --progress?

**Comment by mtnjustme** [comment]
> 
> 
> > The dedupe command does support the rc flag, however, it does a lot of stuff before starting to report that it checked/deleted files, and until then the rc stats are 0 checks/deletes, would be nice to have a more live-like rc regarding the dedupe command, also the -P is useless most of the time with dedupe (until it starts to report checks/deletes, which it doesn't at first, but only as it's about to finish).
> 
> I guess I've always thought of dedupe as more of an interactive tool...
> 
> If it counted up directory listing entries would that be useful? As that is what it is doing at the start.

<< It would be useful to have a live session in the rc, not just counting at the beginning and then 20 minutes of silence.
<< it would be a good start though

> 
> > Also it's possible to make dedupe a flag in functions like copy/sync/move, either that or making it possible to copy/sync/move folders that share the same name.
> 
> I'm pretty sure there is an issue about that already...
> 
> > also possible to make a new variable "Merges" for example...
> 
> Do you mean a statistic to show in --progress?

<< Yes, and a stat in rc

**Comment by mtnjustme** [comment]
also when using a large remote it can stay idle without seemingly doing anything, no idea whats the progress, how long left, no reports of anything, if it doesn't merge any folders, deleting duplicates.

**Comment by ncw** [comment]
> also when using a large remote it can stay idle without seemingly doing anything, no idea whats the progress, how long left, no reports of anything, if it doesn't merge any folders, deleting duplicates.

I think this would be solved if we introduced a new "Listing" stat which has been discussed elsewhere. Then you could use --progress and see "Listing" incrementing.

**Comment by mtnjustme** [comment]
> 
> 
> > also when using a large remote it can stay idle without seemingly doing anything, no idea whats the progress, how long left, no reports of anything, if it doesn't merge any folders, deleting duplicates.
> 
> I think this would be solved if we introduced a new "Listing" stat which has been discussed elsewhere. Then you could use --progress and see "Listing" incrementing.

that would be good, also the renames in the rename mode aren't increasing as a stat, renames and merges should be seen in progress, and yea that's all basically...

**Comment by ncw** [comment]
@mtnjustme can you make a new issue with suggestions for `Listing`, `Renames` and `Merges` as new stat values please?

**Comment by mtnjustme** [comment]
> 
> 
> @mtnjustme can you make a new issue with suggestions for `Listing`, `Renames` and `Merges` as new stat values please?

https://github.com/rclone/rclone/issues/4684
